module/Activity/src/Activity/Service/Activity.php

```diff
@@ -110,12 +111,30 @@ public function createActivity(array $params)
         // Find the creator
         $user = $this->getServiceManager()->get('user_role');
         if ($user === 'guest') {
-            throw new \InvalidArgumentException('Guests can not create activities');
+            throw new NotAllowedException('Guests can not create activities');
         }
-        $params['creator'] = $user;
-
         $activity = new ActivityModel();
-        $activity->create($params);
+        $activity->setBeginTime(new \DateTime($params['beginTime']));
+        $activity->setEndTime(new \DateTime($params['endTime']));
+        $activity->setName($params['name']);
+        $activity->setNameEn($params['name_en']);
+        $activity->setLocation($params['location']);
+        $activity->setLocationEn($params['location_en']);
+
+        if (!$params['costs_unknown']) {
+            $activity->setCosts($params['costs']);
+            $activity->setCostsEn($params['costs_en']);
+        }
+
+        $activity->setDescription($params['description']);
+        $activity->setDescriptionEn($params['description_en']);
+        $activity->setCanSignUp($params['canSignUp']);
+
+        // Not user provided input
+        $activity->setCreator($user);
```

Comment:
That is because this is not the part of the code where weird things happen, and because I have hacked around it. The problem occurs at the following line:

https://github.com/GEWIS/gewisweb/pull/268/files#diff-5f81752c423c0c2d5ddbabad479f495dR346

Why is the ->getLidNr() needed?